spec/rspec/snapshot/matchers_spec.rb

```diff
@@ -26,13 +32,122 @@
     </html>
     HTML
 
-    expect(html).to match_snapshot("snapshot/html")
+    expect(html).to match_snapshot('snapshot/html')
   end
 
-  context "when snapshotting non-string objects" do
-    it "stringifies simple POROs" do
-      simple_data_structure = { a_key: %w(some values) }
-      expect(simple_data_structure).to match_snapshot("snapshot/simple_data_structure")
+  context 'when snapshotting non-string objects' do
+    it 'stringifies simple POROs' do
+      simple_data_structure = { a_key: %w[some values] }
+      expect(simple_data_structure).to match_snapshot('snapshot/simple_data_structure')
+    end
+  end
+
+  describe 'configuration options' do
+    describe 'snapshot_dir' do
+      before do
+        @original_dir = RSpec.configuration.snapshot_dir
+        RSpec.configuration.snapshot_dir = 'custom_snap_dir'
+      end
+
+      after do
+        RSpec.configuration.snapshot_dir = @original_dir
+      end
+
+      it 'supports saving snapshots to a specific directory' do
+        json = JSON.pretty_generate(who: 'first', what: 'second')
+        # notice that the fixture actually lives in /custom_snap_dir
+        expect(json).to match_snapshot('snapshot/in_a_custom_directory')
+      end
+    end
+
+    describe 'snapshot_extension' do
+      before do
+        @original_ext = RSpec.configuration.snapshot_extension
+        RSpec.configuration.snapshot_extension = 'json'
+      end
+
+      after do
+        RSpec.configuration.snapshot_extension = @original_ext
+      end
+
+      it 'supports setting other extensions' do
+        json = JSON.pretty_generate(danger: 'will robinson')
+        # notice that the snapshot extension is actually .json instead of .snap
```

Comment:
TODO: actually test that the file is written to this location (you know, a test)